Library/Homebrew/install.rb

```diff
@@ -0,0 +1,51 @@
+require "diagnostic"
+require "fileutils"
+require "hardware"
+require "development_tools"
+
+module Homebrew
+  module Install
+    module_function
+
+    def check_ppc
+      case Hardware::CPU.type
+      when :ppc
+        abort <<~EOS
+          Sorry, Homebrew does not support your computer's CPU architecture.
+          For PPC support, see: https://github.com/mistydemeo/tigerbrew
+        EOS
+      end
+    end
+
+    def check_writable_install_location
+      raise "Cannot write to #{HOMEBREW_CELLAR}" if HOMEBREW_CELLAR.exist? && !HOMEBREW_CELLAR.writable_real?
+      raise "Cannot write to #{HOMEBREW_PREFIX}" unless HOMEBREW_PREFIX.writable_real? || HOMEBREW_PREFIX.to_s == "/usr/local"
```

Comment:
Can you make both the above block `if`s (and refactor the `unless .. ||`). I know not your code but nice to refactor this so it's not wider than the GitHub diff UI.